vss-processor/src/test/kotlin/org/eclipse/kuksa/vssprocessor/parser/YamlDefinitionParserTest.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.kuksa.vssprocessor.parser
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import java.io.File
+
+class YamlDefinitionParserTest : BehaviorSpec({
+    given("A parser for yaml files") {
+        val parser = YamlDefinitionParser()
+        val classLoader = parser::class.java.classLoader!!
+
+        and("a specification file of version 4") {
+            val resourceUrl = classLoader.getResource("vss_rel_4.0.yaml")!!
+            val specificationFile = File(resourceUrl.path)
+
+            `when`("parsing the file") {
+                val parsedSpecifications = parser.parseSpecifications(specificationFile)
+
+                then("the correct number of specification models should be parsed") {
+                    parsedSpecifications.size shouldBe 1197
```

Comment:
Did you count them or did you just check how many came out after this test failed? ;)

Reply:
<img width="394" alt="Screenshot 2023-10-05 at 11 35 44" src="https://github.com/eclipse-kuksa/kuksa-android-sdk/assets/1355532/ae4e2c8a-67ca-431f-873f-8af6d597bd5f">

I did this. :D Also the reason why I found a bug where the last element was missing.

Reply:
You should add that as a note next to the number, otherwise in case the specification gets updated people might easily come into the situation where they have to update the number and just use the result from the failing test instead of validating the right number :)